File: cpp/LeetCode3423.cpp
class Solution {
public:
    int maxAdjacentDistance(vector<int>& nums) {
        int max_ = std::abs(nums[0] - nums[nums.size()-1]);

        for (int i = 0; i < nums.size()-1; i++) {
            int diff = std::abs(nums[i] - nums[i+1]);
            if (diff > max_) {
                max_ = diff;
            }
        }
        return max_;
    }
};

File: python/LeetCode3423.py
class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        max_ = abs(nums[0] - nums[-1])
        for i in range(len(nums) - 1):
            diff = abs(nums[i] - nums[i + 1])
            if diff > max_:
                max_ = diff

        return max_
